lightning/src/ln/channelmanager.rs

```diff
@@ -913,23 +913,30 @@ macro_rules! handle_chan_restoration_locked {
 			}
 
 			let funding_broadcastable: Option<Transaction> = $funding_broadcastable; // Force type-checking to resolve
-			macro_rules! handle_cs { () => {
-				if let Some(monitor_update) = chanmon_update {
-					assert!($order == RAACommitmentOrder::RevokeAndACKFirst);
-					// We only ever broadcast a funding transaction in response to a funding_signed
-					// message and the resulting monitor update. Thus, on channel_reestablish
-					// message handling we can't have a funding transaction to broadcast. When
-					// processing a monitor update finishing resulting in a funding broadcast, we
-					// cannot have a second monitor update, thus this case would indicate a bug.
-					assert!(funding_broadcastable.is_none());
-					// Given we were just reconnected or finished updating a channel monitor, the
-					// only case where we can get a new ChannelMonitorUpdate would be if we also
-					// have some commitment updates to send as well.
-					assert!($commitment_update.is_some());
-					if let Err(e) = $self.chain_monitor.update_channel($channel_entry.get().get_funding_txo().unwrap(), monitor_update) {
-						break handle_monitor_err!($self, e, $channel_state, $channel_entry, RAACommitmentOrder::CommitmentFirst, false, true);
+			if let Some(monitor_update) = chanmon_update {
```

Comment:
Sorry I missed it, what was the reasoning for adding this commit: e6b1635f07d159da565d1bc5baecfbe165c8c26b

Reply:
See discussion at https://github.com/rust-bitcoin/rust-lightning/pull/851#discussion_r632094939